src/Formats/ProtobufSerializer.cpp

```diff
@@ -298,7 +299,10 @@ namespace
             try
             {
                 /// TODO: use accurate::convertNumeric() maybe?
-                result = boost::numeric_cast<DestType>(value);
+                if constexpr (std::is_same_v<SrcType, IPv4>)
+                    result = boost::numeric_cast<DestType>(value.toUnderType());
```

Comment:
Looks strange, why do we need that?

Reply:
To support type `IPv4`

Reply:
We can write totally separate serializer for `IPv4` without reusing `ProtobufSerializerNumber`. But I am not sure that it's worth it

Reply:
No, I think the current approach is ok. I'd try to make `DataTypeIPv4` a derived class from `DataTypeNumberBase<UInt32>` in order to get rid of `toUnderType()` here, but that's of course optional and not directly linked to this PR.